frontend/public/components/secrets/create-secret/utils.ts

```diff
@@ -94,3 +97,38 @@ export const useSecretDescription = (typeAbstraction: SecretTypeAbstraction): st
       return null;
   }
 };
+
+export const opaqueEntriesToObject = (opaqueEntriesArray: OpaqueDataEntry[]): SecretStringData => {
+  return _.reduce(
+    opaqueEntriesArray,
+    (acc, entry) =>
+      _.assign(acc, {
+        [entry?.key]: entry?.isBase64 || entry?.isBinary ? entry.value : Base64.encode(entry.value),
+      }),
+    {},
+  );
+};
+
+export const newOpaqueSecretEntry = (): OpaqueDataEntry => {
+  return {
+    key: '',
+    value: '',
+    uid: _.uniqueId(),
+  };
+};
+
+export const opaqueSecretObjectToArray = (stringData: SecretStringData): OpaqueDataEntry[] => {
+  if (_.isEmpty(stringData)) {
+    return [newOpaqueSecretEntry()];
+  }
+  return _.map(stringData, (value, key) => {
+    const isValueBinary = isBinary(null, Buffer.from(value, 'base64'));
```

Comment:
I don't know that this check is correct. "stringData" should not be base64 encoded.
```suggestion
    const isValueBinary = isBinary(null, Buffer.from(value));
```

Reply:
Yes it shouldn't be as you said, the problem was that the `DroppableFileInput` component passes an encoded string `fileData` into its `onChange` callback, when a binary file gets uploaded. This is something I have not really identified previously, but in the current update I have made some logic adjustment to this.